packages/pglite/src/base.ts

```diff
@@ -156,47 +188,73 @@ export abstract class BasePGlite
    */
   async #runQuery<T>(
     query: string,
-    params?: any[],
+    params: any[] = [],
     options?: QueryOptions,
   ): Promise<Results<T>> {
     return await this._runExclusiveQuery(async () => {
       // We need to parse, bind and execute a query with parameters
       this.#log('runQuery', query, params, options)
       await this._handleBlob(options?.blob)
-      const parsedParams =
-        params?.map((p) => serializeType(p, options?.setAllTypes)) || []
+
       let results
+
       try {
+        const parseResults = await this.#execProtocolNoSync(
+          serializeProtocol.parse({ text: query, types: options?.paramTypes }),
+          options,
+        )
+
+        const dataTypeIDs = parseDescribeStatementResults(
+          (
+            await this.#execProtocolNoSync(
+              serializeProtocol.describe({ type: 'S' }),
```

Comment:
does this mean we have one additional roundtrip per query?

Reply:
Yes, extra call to PG, I think it's very fast. I will check...

Reply:
FWIW I've run benchmarks against all the popular JS postgres libraries, and Postgres.js was by far the fastest for my workload despite doing it this way. I think there are other things holding those libraries back, like inefficient use of connections.

Reply:
Yep, this query takes ~0.3ms, not worth worrying about.